### PR TITLE
Fixing ingress spec when a TLS secret name is used

### PR DESCRIPTION
**Pull request by cccs-eric**
### What does this PR do?

When chePluginRegistryIngressSecretName is set, the resulting spec is invalid and deployment is not possible.  It was related to a wrong indentation of the YAML.  I also refactored the way the hostname was generated, which is now in-line with the che-devfile-registry code.

### What issues does this PR fix or reference?
I haven't created an issue for this.

### How to test this PR?
Simply trying to deploy with helm will fail:
`helm upgrade --install che-plugin-registry --namespace che --set global.ingressDomain=my.domain.com --set chePluginRegistryImage=path.to.cr/che-plugin-registry:latest --set chePluginRegistryIngressSecretName=che-tls ./deploy/kubernetes/che-plugin-registry/`

Results in this error:
`Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(Ingress.spec.rules[0]): unknown field "tls" in io.k8s.api.extensions.v1beta1.IngressRule
`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by ericwill** [comment]
I'm a bit out of my element here, @tolusha or @skabashnyuk could someone from your team(s) review this one?

**Comment by tolusha** [comment]
@cccs-eric 
Did you forget to commit `values.yaml` ?

**Comment by cccs-eric** [comment]
@tolusha 
`values.yaml` does not change since the problem was in the spec generated by `ingress.yaml`.  When `.Values.chePluginRegistryIngressSecretName` is not empty, a `tls` spec will be generated and its indentation is wrong.  So kubernetes will complain about an invalid spec.

**Comment by tolusha** [comment]
@cccs-eric 
I got the problem with a wrong indentation.
I am just wondering how `Values.hostnameOverride` is used.

**Comment by cccs-eric** [comment]
Sorry, I see what you mean now.  I added a line in `values.yaml`, which is very similar to what we have in [che-devfile-registry](https://github.com/eclipse/che-devfile-registry/blob/master/deploy/kubernetes/che-devfile-registry/values.yaml#L14).  The idea was to make them both look the same...

**Review by tolusha — COMMENTED** [review]
@cccs-eric 
To be consistent with templates in `eclipse/che` repository have a look at my proposals please.

**Comment by cccs-eric** [comment]
@tolusha 
I'm fine with the changes you proposed.

File: deploy/kubernetes/che-plugin-registry/templates/ingress.yaml [hunk 1]
```diff
@@ -15,16 +15,16 @@ metadata:
...
           serviceName: che-plugin-registry
           servicePort: 8080
 {{- if .Values.chePluginRegistryIngressSecretName }}
-    tls:
-    - hosts:
-      - {{ printf "che-plugin-registry-%s.%s" .Release.Namespace .Values.global.ingressDomain }}
-      secretName: {{ .Values.chePluginRegistryIngressSecretName }}
+  tls:
+  - hosts:
+    - {{ template "chePluginRegistryHost" . }}
```
Comment (tolusha) [inline deploy/kubernetes/che-plugin-registry/templates/ingress.yaml:28]:
```suggestion
    - {{ template "pluginRegistryHost" . }}
```

Comment (tolusha) [reply deploy/kubernetes/che-plugin-registry/templates/ingress.yaml:28]: One litter remark.

**Review by tolusha — APPROVED** [review]

File: deploy/kubernetes/che-plugin-registry/templates/ingress.yaml (hunk 1 above)
Comment (cccs-eric) [reply deploy/kubernetes/che-plugin-registry/templates/ingress.yaml:28]: Good catch

**Pull request merged**